src/cluster_migrateslots.c

```diff
@@ -0,0 +1,2119 @@
+ *          ┌──────────────▼─────────────┐  │
+ *          │SLOT_IMPORT_RECEIVE_SNAPSHOT┼──┤
+ *          └──────────────┬─────────────┘  │
+ *             SNAPSHOT-EOF│                │
+ *         ┌───────────────▼──────────────┐ │
+ *         │SLOT_IMPORT_WAITING_FOR_PAUSED┼─┤
+ *         └───────────────┬──────────────┘ │
+ *                   PAUSED│                │
+ *         ┌───────────────▼──────────────┐ │ Error Conditions:
+ *         │SLOT_IMPORT_FAILOVER_REQUESTED┼─┤  1. OOM
+ *         └───────────────┬──────────────┘ │  2. Slot Ownership Change
+ *         FAILOVER-GRANTED│                │  3. Demotion to replica
+ *          ┌──────────────▼─────────────┐  │  4. FLUSHDB
+ *          │SLOT_IMPORT_FAILOVER_GRANTED┼──┤  5. Connection Lost
+ *          └──────────────┬─────────────┘  │  6. No ACK from source (timeout)
+ *       Takeover Performed│                │
+ *          ┌──────────────▼───────────┐    │
+ *          │SLOT_MIGRATION_JOB_SUCCESS┼──-─┤
+ *          └──────────────────────────┘    │
+ *                                          │
+ *    ┌─────────────────────────────────────▼─┐
+ *    │SLOT_IMPORT_FINISHED_WAITING_TO_CLEANUP│
+ *    └────────────────────┬──────────────────┘
+ * Unowned Slots Cleaned Up│
+ *           ┌─────────────▼───────────┐
+ *           │SLOT_MIGRATION_JOB_FAILED│
+ *           └─────────────────────────┘
+ *
+ */
+
+int clusterIsSlotImporting(int slot) {
+    listNode *ln;
+    listIter li;
+    listRewind(server.cluster->slot_migration_jobs, &li);
+    while ((ln = listNext(&li)) != NULL) {
+        slotMigrationJob *job = ln->value;
+        if (job->type != SLOT_MIGRATION_IMPORT) continue;
+        if (isSlotMigrationJobFinished(job)) continue;
+        if (isSlotInSlotRanges(slot, job->slot_ranges)) return 1;
+    }
+    return 0;
+}
+
+/* Sent by the source to the target to initiate the AOF formatted snapshot.
+ * Note that if there is an error in the request, we send a fail message in
+ * order to prevent infinite retry in the case of incompatibility.
+ *
+ * CLUSTER SYNCSLOTS ESTABLISH is the only CLUSTER SYNCSLOTS subcommand that
+ * will return a reply. Errors are written from the perspective of the end user
+ * to help with debugging migrations. */
+void clusterCommandSyncSlotsEstablish(client *c) {
+    char *name = NULL;
+    clusterNode *source_node = NULL;
+    clusterNode *owning_node = NULL;
+    list *slot_ranges = NULL;
+
+    if (!nodeIsPrimary(server.cluster->myself)) {
+        addReplyError(c, "Target node is not a primary");
+        return;
+    }
+
+    if (isAnySlotInManualImportingState() ||
+        isAnySlotInManualMigratingState()) {
+        addReplyError(c, "A slot on the target node is being manually imported "
+                         "or migrated");
+        return;
+    }
+
+    if (c->slot_migration_job) {
+        addReplyError(c,
+                      "Slot migration client is already a slot migration job");
```

Comment:
super nit. are you using a width of 80 col :)? the line break stands out a bit 
```suggestion
        addReplyError(c, "Slot migration client is already a slot migration job");
```

Reply:
Yeah I am trying to get to 80 - it helps with reading on mobile device or in Github diffs

I would say le'ts not argue about line breaks unless we are going to enforce them in clang-format?